pytest_ethereum/utils/linker.py

```diff
@@ -0,0 +1,96 @@
+from eth_utils import to_canonical_address, to_hex
+from eth_utils.toolz import assoc_in
+
+from ethpm.utils.chains import (
+    check_if_chain_matches_chain_uri,
+    create_block_uri,
+    get_chain_id,
+)
+from pytest_ethereum.exceptions import LinkerError
+
+
+def pluck_matching_uri(deployment_data, w3):
+    """
+    Return any blockchain uri that matches w3-connected chain, if one
+    is present in the deployment data keys.
+    """
+    for uri in deployment_data.keys():
+        if check_if_chain_matches_chain_uri(w3, uri):
+            return uri
+    raise LinkerError(
+        "No matching blockchain URI found in deployment_data: {0}, for w3 instance: {1}.".format(
+            list(deployment_data.keys()), w3.__repr__()
+        )
+    )
+
+
+def contains_matching_uri(deployment_data, w3):
+    """
+    Returns true if any blockchain uri in deployment data matches
+    w3-connected chain.
+    """
+    for uri in deployment_data.keys():
+        if check_if_chain_matches_chain_uri(w3, uri):
+            return True
+    return False
+
+
+def create_latest_block_uri(w3, tx_receipt):
+    """
+    Creates a new block uri from data in w3 and provided tx_receipt.
+    """
+    chain_id = get_chain_id(w3)
+    block_hash = to_hex(tx_receipt.blockHash)
+    return create_block_uri(to_hex(chain_id), block_hash)
```

Comment:
I don't think this is right (using the `chain_id`).  I think that should be the block hash of the genesis block.

http://ethpm.github.io/ethpm-spec/package-spec.html#bip122-uris

Reply:
`get_chain_id()` is a util from `py-ethpm`
```python
def get_chain_id(web3: Web3) -> str:
    return web3.eth.getBlock(0)["hash"]
```
Which seems to me like returns the genesis block hash - maybe it's just a poorly name util? `get_genesis_block_hash`?

Reply:
Yeah, `chain_id` is a whole different concept and renaming will help prevent that mixup.